code/modules/mob/living/silicon/robot/subtypes/thinktank/thinktank_interactions.dm

```diff
@@ -0,0 +1,57 @@
+/mob/living/silicon/robot/platform/attack_hand(mob/user)
+
+	if(!opened)
+		if(recharging)
+			recharging.dropInto(loc)
+			user.put_in_hands(recharging)
+			user.visible_message(SPAN_NOTICE("\The [user] pops \the [recharging] out of \the [src]'s recharging port."))
+			recharging = null
+			return TRUE
+		if(length(stored_atoms))
+			var/weakref/remove_ref = stored_atoms[length(stored_atoms)]
+			var/atom/movable/removing = remove_ref?.resolve()
+			if(!istype(removing) || QDELETED(removing) || removing.loc != src)
+				LAZYREMOVE(stored_atoms, remove_ref)
+			else
+				user.visible_message(SPAN_NOTICE("\The [user] begins unloading \the [removing] from \the [src]'s cargo compartment."))
+				if(do_after(user, 3 SECONDS, src) && !QDELETED(removing) && removing.loc == src)
+					drop_stored_atom(removing, user)
+			return TRUE
+
+	. = ..()
+
+/mob/living/silicon/robot/platform/attackby(obj/item/W, mob/user)
+
+	if(istype(W, /obj/item/weapon/cell) && !opened)
+		if(recharging)
+			to_chat(user, SPAN_WARNING("\The [src] already has a cell inserted into its recharging port."))
+		else if(user.unEquip(W, src))
+			recharging = W
+			recharge_complete = FALSE
+			user.visible_message(SPAN_NOTICE("\The [user] slots \the [recharging] into \the [src]'s recharging port."))
+		return TRUE
+
+	if(istype(W, /obj/item/device/floor_painter))
+		return FALSE // Paint sprayer wil call try_paint() in afterattack()
+
+	. = ..()
+
+/mob/living/silicon/robot/platform/attack_ghost(mob/observer/ghost/user)
```

Comment:
If these are mapped in, could ghosts activate them and start tooling around before round-start (Possibly even before initializations are done)?

Reply:
Not sure. It hasn't been an issue with attack_ghost stuff on Bay to my knowledge but I can put in an initialization check if you like.

Reply:
I can give it a try on a live test

Reply:
Added a ticker check to attack_ghost so should be sorted now either way.